iris-mpc-cpu/src/execution/hawk_main/rot.rs

```diff
@@ -0,0 +1,77 @@
+use std::ops::Deref;
+
+use iris_mpc_common::ROTATIONS;
+use itertools::Itertools;
+
+/// VecRots are lists of things for each rotation.
+pub struct VecRots<R> {
+    pub rotations: Vec<R>,
+}
+
+impl<R> Deref for VecRots<R> {
+    type Target = Vec<R>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.rotations
+    }
+}
+
+impl<R> VecRots<R> {
+    pub fn new(rotations: Vec<R>) -> Self {
+        assert_eq!(rotations.len(), ROTATIONS);
+        Self { rotations }
+    }
+
+    /// Get the item attached to the center rotation.
+    pub fn center(&self) -> &R {
+        &self.rotations[self.rotations.len() / 2]
+    }
+
+    /// Get the item attached to the center rotation.
+    pub fn into_center(self) -> R {
+        let middle = self.rotations.len() / 2;
+        self.rotations.into_iter().nth(middle).unwrap()
+    }
+
+    /// Flatten a batch of something with rotations into a concatenated Vec.
+    pub fn flatten(batch: &[VecRots<R>]) -> Vec<R>
```

Comment:
to save a copy here it might be enough to return an `impl Iterator<Item = &R>` and change the receiver to also take this signature?
Just a suggestion, idk if this is really needed for perf reasons yet.

Reply:
I am assuming all this logistics around requests/eyes/rotations/sessions does not matter much for performance compared to the MPC. Although if that turns out worthwhile we can hunt for unnecessary allocations, copies, atomic increments, etc through the whole pipeline.